mysql-test/t/bug90351.test

```diff
@@ -0,0 +1,29 @@
+#
+# Bug 90351: status counters decrease on thread destruction
+#
+--source include/have_innodb.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+--source include/count_sessions.inc
+
+connect (con1,localhost,root,,);
+
+connection con1;
+SET DEBUG_SYNC="thd_cleanup_start WAIT_FOR increased_counter";
```

Comment:
Now the testcase syncs does not make sense: so this thread, upon reaching thd_cleanup_start, waits for increased_counter. But nothing signals it.

Reply:
I could signal it from a different connection with a dummy statemnet or something - but I found that its not needed.

* With the unmodified code, the thread waits here, and the assertion is hit
* With the modified code, the test case passes, the thread ends even if I do not send a signal for this sync point.

Reply:
I see, but the testcase needs all connections actually closed - so do you rely on debug_sync timeout then?